File: DevEnvironmentCLion/Base/core/Threads.cpp
#include "PreCompiled.h"
#include "Threads.h"

using namespace base;

/**
 * Sleeps the current thread.
 */
void Threads::sleep(size_t milliseconds)
{
    std::this_thread::sleep_for(std::chrono::milliseconds(milliseconds));
}

File: DevEnvironment/Base/src/main/java/base/core/Threads.java
package base.core;

import base.BaseException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * The Threads class implements complementary APIs for threads.
 */
public final class Threads {
    private static final Logger LOG = LoggerFactory.getLogger(Threads.class);

    /**
     * Sleeps the current thread.
     */
    public static void sleep(int milliseconds) {
        try {
            Thread.sleep(milliseconds);
        }
        catch (InterruptedException exception) {
            String errorMessage =
                "The current thread failed sleeping for: " + milliseconds + " milliseconds " +
                ", due to the following error: " + exception.getMessage();

            LOG.error(errorMessage);
            throw new BaseException(errorMessage);
        }
    }

    /**
     * Disables the default constructor - This is a static class.
     */
    private Threads() {
    }
}
